transaction-view/src/entry_view.rs

```diff
@@ -0,0 +1,340 @@
+use {
+    crate::{
+        bytes::check_remaining, result::Result, transaction_data::TransactionData,
+        transaction_frame::TransactionFrame, transaction_view::UnsanitizedTransactionView,
+    },
+    bytes::{Buf, Bytes},
+    solana_sdk::hash::Hash,
+};
+
+pub trait EntryData: Clone {
+    /// Associated type for the transaction data.
+    type Tx: TransactionData;
+
+    /// Get the remaining data slice for the entry.
+    fn data(&self) -> &[u8];
+
+    /// Move offset forward `num_bytes`.
+    fn advance(&mut self, num_bytes: usize);
+
+    /// Split the a `TransactionData` off the front of the entry.
+    fn split_to(&mut self, index: usize) -> Self::Tx;
+}
+
+/// Reads an entry from the provided `data`.
+/// The `handle_transaction` closure is called for each transaction in the
+/// entry.
+/// Returns the number of hashes, the hash, and the number of transactions in
+/// the entry.
+/// If the entry, or any transactions within the entry, are invalid, an error
+/// is returned.
+pub fn read_entry<D: EntryData>(
+    data: &mut D,
+    mut handle_transaction: impl FnMut(UnsanitizedTransactionView<D::Tx>),
```

Comment:
this may look odd, and I'll explain my thinking around this.

1. For testing it makes sense to operate on and collect `UnsanitizedTransactionView`. We do not need/want to test the sanitization code here.
2. We want flexibility to use a pre-allocated array here when possible
3. We may not even want to collect these into a Vec; it's possible that we just immediately pass these to some transaction handling fn (sanitizing and inserting to replay scheduler with no intermediate allocation)

Reply:
This also seems like a good point to clarify that this also gives us the opportunity to make sanitization checks parallel.

The initial parsing (w/o sanitization) is not parallelizable since we cannot know where the boundaries between transactions are without parsing the actual data.